homeassistant/components/modbus/binary_sensor.py

```diff
@@ -23,27 +38,62 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Modbus binary sensors."""
-    sensors = []
 
     if discovery_info is None:  # pragma: no cover
         return
 
+    sensors: list[ModbusBinarySensor | SlaveSensor] = []
+    hub = get_hub(hass, discovery_info[CONF_NAME])
     for entry in discovery_info[CONF_BINARY_SENSORS]:
-        hub = get_hub(hass, discovery_info[CONF_NAME])
-        sensors.append(ModbusBinarySensor(hub, entry))
-
+        slave_count = entry.get(CONF_SLAVE_COUNT, 0)
+        sensor = ModbusBinarySensor(hub, entry, slave_count)
+        if slave_count > 0:
+            sensors.extend(await sensor.async_setup_slaves(hass, slave_count, entry))
+        sensors.append(sensor)
     async_add_entities(sensors)
 
 
 class ModbusBinarySensor(BasePlatform, RestoreEntity, BinarySensorEntity):
     """Modbus binary sensor."""
 
+    def __init__(self, hub: ModbusHub, entry: dict[str, Any], slave_count: int) -> None:
+        """Initialize the Modbus binary sensor."""
+        self._count = slave_count + 1
+        self._coordinator: DataUpdateCoordinator[Any] | None = None
```

Comment:
Because the coordinator is managed by the `ModbusBinarySensor`, if this entity is disabled, none of the slaves would work.

I thought this entity would also listen to the update coordinator.

Reply:
No this class does the actual reading. This is because the majority of configurations will not use slaves.

And it is correct that if an instance of ModbusBinarySensor fails or are disabled so should all slaves belonging to that  particular instance.